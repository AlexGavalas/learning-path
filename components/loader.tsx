export const Loader = () => (
    <div className="animate-loader bg-no-repeat bg-center bg-gradient-to-r from-white via-teal-500 dark:via-yellow-500 to-white bg-[length:25%_100%] w-full h-[2px] absolute bottom-[-8px]" />
);
