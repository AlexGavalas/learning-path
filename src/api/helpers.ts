import axios from 'axios';

export const fetchFileFromStorage = async (
    filename: string,
): Promise<string> => {
    const fileServerUrl = String(process.env.PUBLIC_FILE_SERVER_URL);

    const response = await axios.get(`${fileServerUrl}/${filename}`, {
        responseType: 'text',
    });

    return response.data as string;
};
