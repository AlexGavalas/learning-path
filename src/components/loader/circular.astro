<style>
    @keyframes spin {
        to {
            transform: rotate(360deg);
        }
    }

    @keyframes spin-offset {
        0% {
            stroke-dasharray: 0, 628;
        }
        50% {
            stroke-dasharray: 314;
        }
        100% {
            stroke-dasharray: 628, 0;
        }
    }

    .spin {
        stroke-dasharray: 0, 628; /* 2 * PI * r */
        transform-origin: center;
        animation:
            spin-offset 2s ease-in-out infinite alternate,
            spin 2s linear infinite;
    }
</style>

<svg viewBox="0 0 800 800" xmlns="http://www.w3.org/2000/svg">
    <circle
        class="spin"
        cx="400"
        cy="400"
        fill="none"
        r="200"
        stroke="#f1d20d"
        stroke-width="50"
        stroke-linecap="round"
    >
    </circle>
</svg>
