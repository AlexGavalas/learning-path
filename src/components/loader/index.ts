export { Loader } from './loader';
export { CicurlarLoader } from './circular-loader';
