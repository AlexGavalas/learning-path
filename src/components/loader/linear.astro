---
interface Props {
    id?: string;
}

const { id } = Astro.props;
---

<div
    id={id}
    class="h-full animate-loader bg-gradient-to-r from-white from-0% via-light-primary via-50% to-white to-100% bg-[length:25%_100%] bg-center bg-no-repeat dark:via-dark-primary"
>
</div>
