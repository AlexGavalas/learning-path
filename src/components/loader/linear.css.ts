import { style } from '@vanilla-extract/css';

import { fullSize } from '../../styles/animations.css';
import { vars } from '../../styles/theme.css';

export const loader = style({
    animation: `${fullSize} 2s linear infinite alternate`,
    backgroundImage: `linear-gradient(to right, white 0%, ${vars.color.primary} 50%, white 100%)`,
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat',
    backgroundSize: '25% 100%',
    height: '100%',
});
