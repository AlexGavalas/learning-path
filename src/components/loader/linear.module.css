.loader {
    animation: fullSize 2s linear infinite alternate;
    background-image: linear-gradient(
        to right,
        oklch(from var(--color-primary) 0.4 c h),
        oklch(from var(--color-primary) 0.8 c h)
    );
    background-position: center;
    background-repeat: no-repeat;
    background-size: 25% 100%;
    height: 100%;
}

@keyframes fullSize {
    to {
        background-size: 100% 100%;
    }
}
