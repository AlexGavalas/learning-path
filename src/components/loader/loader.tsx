export const Loader = () => (
    <div className="absolute bottom-[-8px] h-[2px] w-full animate-loader bg-gradient-to-r from-white via-teal-500 to-white bg-[length:25%_100%] bg-center bg-no-repeat dark:via-yellow-500" />
);
