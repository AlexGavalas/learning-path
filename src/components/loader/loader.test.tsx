import { axe } from 'jest-axe';

import { renderWithUser } from '~test/helpers';
import { Loader } from './loader';

describe('<Loader />', () => {
    it('renders', () => {
        const { container } = renderWithUser(<Loader />);

        expect(container).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const { container } = renderWithUser(<Loader />);

        const a11yResults = await axe(container);

        expect(a11yResults).toHaveNoViolations();
    });
});
