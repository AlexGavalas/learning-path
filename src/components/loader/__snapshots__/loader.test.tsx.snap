// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders 1`] = `
<div>
  <div
    class="absolute bottom-[-8px] h-[2px] w-full animate-loader bg-gradient-to-r from-white via-teal-500 to-white bg-[length:25%_100%] bg-center bg-no-repeat dark:via-yellow-500"
  />
</div>
`;
